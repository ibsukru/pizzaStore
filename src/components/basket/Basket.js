import React from 'react'
import { getPrice, getTotalPrice } from '../../helpers'

const Basket = ({ basket, onPay, onToggleBasket, onRemovingOrder }) => {
  if (basket.completed) {
    return (
      <div>
        <h2>Well done, order is on your way!</h2>
        <div className='basket-actions'>
          <a className='basket-actions__toggle' onClick={onToggleBasket}>
            I want more pizza!
          </a>
        </div>
      </div>
    )
  }

  const { orders } = basket

  if (orders && !orders.length) {
    return (
      <div>
        <h1>
          Your basket is empty, <a onClick={onToggleBasket}>Here</a> you can add items.
        </h1>
      </div>
    )
  }

  return (
    <div>
      <h1>Hey here is your basket, these things are yummy!</h1>
      <ul className='orders'>
        {orders.map((order, index) => {
          return (
            <li className='orders__order' key={order.name + index}>
              {order.name} pizza with
              {order.toppings &&
                order.toppings
                  .filter(t => t.defaultSelected)
                  .map(({ topping }) => {
                    return <span key={topping.name}> {topping.name}, </span>
                  })} costs {getPrice(order).toFixed(2)}$,
              <div><a onClick={() => onRemovingOrder(index)}>Remove</a></div>
            </li>
          )
        })}
      </ul>
      <div className='basket-actions'>
        <a className='basket-actions__pay' onClick={() => onPay(basket)}>
          Finish and pay {getTotalPrice(orders).toFixed(2)}$
        </a>
        <a className='basket-actions__toggle' onClick={onToggleBasket}>
          Add more...
        </a>
      </div>
    </div>
  )
}

export default Basket
