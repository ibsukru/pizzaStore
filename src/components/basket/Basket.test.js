/* eslint-env jest */
import React from 'react'
import { create } from 'react-test-renderer'
import Basket from './Basket'
;[
  {
    basket: {
      orders: [{ name: 'some name' }]
    },
    getPrice: jest.fn(f => 1),
    getTotalPrice: jest.fn(f => 1),
    onPay: f => f,
    onToggleBasket: f => f,
    onRemovingOrder: f => f
  }
].forEach((props) => {
  test(`With props ${JSON.stringify(props)} Should match snapshot...`, () => {
    expect(create(<Basket {...props} />).toJSON()).toMatchSnapshot()

    expect(props.getPrice).toBeCalled()
    expect(props.getTotalPrice).toBeCalled()
  })
})
