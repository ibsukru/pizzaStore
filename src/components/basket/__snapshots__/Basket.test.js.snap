// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With props {"basket":{"orders":[{"name":"some name"}]}} Should match snapshot... 1`] = `
<div>
  <h1>
    Hey here is your basket, these things are yummy!
  </h1>
  <ul
    className="orders"
  >
    <li
      className="orders__order"
    >
      some name
       pizza with
       costs 
      1.00
      $,
      <div>
        <a
          onClick={[Function]}
        >
          Remove
        </a>
      </div>
    </li>
  </ul>
  <div
    className="basket-actions"
  >
    <a
      className="basket-actions__pay"
      onClick={[Function]}
    >
      Finish and pay 
      1.00
      $
    </a>
    <a
      className="basket-actions__toggle"
      onClick={[Function]}
    >
      Add more...
    </a>
  </div>
</div>
`;
